src/buildercore/core.py

```diff
@@ -111,13 +107,22 @@ def connect_aws_with_stack(stackname, service):
     return connect_aws_with_pname(pname, service)
 
 def find_ec2_instance(stackname):
-    "returns ec2 instance data for a *specific* stackname"
+    "returns list of ec2 instances data for a *specific* stackname"
```

Comment:
I recall that in the not-a-cluster case this returns a list of 1 element, which is the instance data; it was already used with `[0]` after the call